File: eRent.Services/DataDB/Rezervacija.cs
using System;
using System.Collections.Generic;

namespace eRent.Services.DataDB;

public partial class Rezervacija
{
    public int RezervacijaId { get; set; }

    public bool? MjesecnaRezervacija { get; set; }

    public DateTime? DatumPocetka { get; set; }

    public DateTime? DatumKraja { get; set; }

    public string? ImePrezime { get; set; }

    public string? BrojTelefona { get; set; }

    public int NekretninaId { get; set; }

    public string? Nazivnekretnine { get; set; }

    public bool? Odobrena { get; set; }

    public bool? Otkazana { get; set; }

    public int? KorisnikId { get; set; }

    public virtual Nekretnina Nekretnina { get; set; } = null!;

    public virtual ICollection<NekretninaRezervacija> NekretninaRezervacijas { get; } = new List<NekretninaRezervacija>();
}

File: mobile/myapp/lib/model/rezervacija.dart
import 'package:json_annotation/json_annotation.dart';

part 'rezervacija.g.dart';

@JsonSerializable()
class Rezervacija {
  int? rezervacijaId;
  int? nekretninaId;
  int? korisnikId;
  bool? mjesecnaRezervacija;
  bool? odobrena;
  DateTime? datumPocetka;
  DateTime? datumKraja;
  String? imePrezime;
  String? brojTelefona;
  String? nazivnekretnine;
  bool? otkazana;
  bool? odbijena;
  Rezervacija();

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);

  get id => null;

  /// Connect the generated [_$ProductToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}
